fix regex for NSX-T TZ creation version check

rexit1982 opened the pull request:
Improper regex was causing both conditions to hit making builds fail.

rexit1982 committed:
fix regex for NSX-T TZ creation version check

rexit1982 committed:
Merge branch 'rutgerblom:dev-v8' into dev-v8

luischanu commented:
I'm not quite sure how both could "hit" given one is the negated version of the other.  Out of curiosity, what version were you checking against?

That said, if this needs to be updated, I would propose that we use the built-in Ansible Version comparison capability rather than RegEx to accomplish this.   Much cleaner and easier to read.

rexit1982 commented:
You are right, I made two changes at once.....I was deploying 4.1.2.5, I
changed the regex and updated my collections at the same time.  Id have to
go look if I still have logs but I think it was actually missing the pre
4.2 then getting an error from my old collection not liking the parameter.
Is there any difference between the two with that one line commented out?

On Wed, Nov 27, 2024, 1:47 AM Luis Chanu ***@***.***> wrote:

> I'm not quite sure how both could "hit" given one is the negated version
> of the other. Out of curiosity, what version were you checking against?
>
> That said, if this needs to be updated, I would propose that we use the
> built-in Ansible Version comparison capability rather than RegEx to
> accomplish this. Much cleaner and easier to read.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rutgerblom/SDDC.Lab/pull/466#issuecomment-2503036618>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABZ32E53E6YB4KZIND5UAGT2CVTGRAVCNFSM6AAAAABSRYQ2J2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKMBTGAZTMNRRHA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

rexit1982 commented:
Going to close this, while I'm not sure the logic was working as intended, I don't think it is actually necessary as the tasks are the same.
